app/models/concerns/arclight/solr_document.rb

```diff
@@ -54,7 +54,8 @@ def repository
     end
 
     def repository_and_unitid
-      [repository, unitid].compact.join(': ')
+      # We need compact_blank here due to https://github.com/projectblacklight/blacklight/issues/2952
+      [repository, unitid].compact_blank.join(': ')
```

Comment:
Is `compact_blank` still needed now that this is fixed: https://github.com/projectblacklight/blacklight/issues/2952